harness/helpers.ts

```diff
@@ -135,17 +136,56 @@ export const wait = (ms: number) => {
     })
 }
 
-export const callOnClientInitialized = async (clientId: string, url: string, callbackURL: string) => {
-    return await fetch(`${url}/client/${clientId}`, {
+export const callOnClientInitialized = async (
+    clientId: string, 
+    proxyUrl: string, 
+    mockServerUrl: string,
+    scope: Scope
+) => {
+    const callbackURL = `/client/${clientId}/onClientInitialized`
+    scope.post(callbackURL).reply(200)
+
+    const response = await fetch(`${proxyUrl}/client/${clientId}`, {
         method: 'POST',
         headers: {
             'Content-Type': 'application/json'
         },
         body: JSON.stringify({
             command: 'onClientInitialized',
             params: [
-                { callbackURL }
+                { callbackURL: `${mockServerUrl}${callbackURL}` }
             ]
         })
     })
+
+    await waitForRequest(
+        scope,
+        callbackURL,
+        3000,
+        'onClientInitialized callback timed out'
+    )
+    
+    return response
 }
+
+export const waitForRequest = async (scope: Scope, callbackURL: string, timeout: number, timeoutMessage: string) => {
+
+    const timeoutPromise = new Promise((_, reject) => {
+        setTimeout(() => {
+            reject(new Error(timeoutMessage))
+        }, timeout)
+    })
+
+    await Promise.race([
+        new Promise((resolve) => {
+            const callback =  (req, inter) => {
+                if (inter.uri === callbackURL) {
```

Comment:
it might be better if this could take the actual interceptor you're waiting rather than just the URL. There may be times where youre using the same URL in two different interceptors but there are other conditions its matching on (headers, body contents etc.)

Reply:
sure, fixed!